lib/docker.ts

```diff
@@ -96,6 +107,50 @@ class Docker {
   }
 
   public async save(targetImage: string, destination: string) {
+    const request = {
+      path: `/images/${targetImage}/get?`,
+      method: "GET",
+      isStream: true,
+      statusCodes: {
+        200: true,
+        400: "bad request",
+        500: "server error",
+      },
+    };
+
+    debug(
+      `Docker.save: targetImage: ${targetImage}, destination: ${destination}`,
+    );
+
+    const modem: Modem = new Modem({ socketPath: this.socketPath });
+
+    return new Promise((resolve, reject) => {
+      modem.dial(request, (err, stream: Stream) => {
+        if (err) {
+          return reject(err);
+        }
+
+        const writeStream = createWriteStream(destination);
+        writeStream.on("error", (err) => {
+          reject(err);
+        });
+        writeStream.on("finish", () => {
+          resolve();
+        });
+
+        stream.on("error", (err) => {
+          reject(err);
+        });
+        stream.on("end", () => {
+          writeStream.end();
+        });
+
+        stream.pipe(writeStream);
+      });
+    });
+  }
+
+  public async saveCli(targetImage: string, destination: string) {
```

Comment:
is anyone using the old implementation?

Reply:
No

Reply:
maybe worth deleting it altogether then?